TBDataPreparation/202108_SPS/scripts/PhysicsEvent.h

```diff
@@ -85,47 +82,47 @@ class Event{
     //
     int SPMT1, SPMT2, SPMT3, SPMT4, SPMT5, SPMT6, SPMT7, SPMT8;
     int CPMT1, CPMT2, CPMT3, CPMT4, CPMT5, CPMT6, CPMT7, CPMT8;
-    
-    UShort_t SiPMHighGain[320];
-    UShort_t SiPMLowGain[320];
 
-    void calibrate(const SiPMCalibration&, EventOut*);
-    void calibratePMT(const PMTCalibration&, EventOut*);
-    	
+    int SiPMHighGain[320];
+    int SiPMLowGain[320];
+
+    void calibrate(const SiPMCalibration&, EventOut*) const;
+    void calibratePMT(const PMTCalibration&, EventOut*) const;
 };
 
-void Event::calibrate(const SiPMCalibration& calibration, EventOut* evout){
+void Event::calibrate(const SiPMCalibration& calibration, EventOut* evout)const{
 
     //SiPM calibration
     //
-    int ccount = 0;
-    int scount = 0;
+    evout->totSiPMCene = 0.;
+    evout->totSiPMSene = 0.;
```

Comment:
Are we sure we want to set to zero the energy scale of SiPMs INSIDE the calibration method? In my view, the two things (setting energy to zero and calibrating signals) should be independent and under user control. In this way, it is a bit counterintuitive even if not wrong.

Reply:
I was thinking to obscure everything inside class methods so that the final user has only to call them but i agree that probably we should implement some setters/getters to give more control to the user.

Reply:
Yes, that is what I was thinking. Let's keep the calibration function(s) doing the conversion from raw_data to GeV. We will add a dedicated method for setting the energy scale to zero. Users might want to use it at any point of the analysis without calling the calibration functions.